projects/schematics/src/migrations/4_0/constructor-deprecations/data/logout.guard.migration.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ANGULAR_ROUTER,
+  AUTH_REDIRECT_SERVICE,
+  AUTH_SERVICE,
+  CMS_SERVICE,
+  LOGOUT_GUARD,
+  PROTECTED_ROUTES_SERVICE,
+  ROUTER,
+  SEMANTIC_PATH_SERVICE,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const LOGOUT_GUARD_CONSTRUCTOR_MIGRATION: ConstructorDeprecation = {
+  // projects/storefrontlib/src/cms-components/user/logout/logout.guard.ts
+  class: LOGOUT_GUARD,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [
+    {
+      className: AUTH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: CMS_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: SEMANTIC_PATH_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: PROTECTED_ROUTES_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: ROUTER,
+      importPath: ANGULAR_ROUTER,
+    },
+    { className: AUTH_REDIRECT_SERVICE, importPath: SPARTACUS_CORE },
```

Comment:
We don't want this variant of constructor, but the one without AuthRedirectService. Remove this parameter and it will be good.